pkg/core/provider.go

```diff
@@ -120,3 +157,80 @@ type providerOption struct {
 }
 
 type ProviderOption func(option *providerOption)
+
+type Sha256SumsEntry struct {
+	Sum      []byte
+	FileName string
+}
+
+// Strings returns the Sha256SumsEntry as a string similar to the sha256 GNU coreutils tool
+func (s *Sha256SumsEntry) String() string {
+	return fmt.Sprintf("%x %s", s.Sum, s.FileName)
+}
+
+// NewSha256SumsEntry parses a Sha256SumsEntry from a line as found in the *_SHA256SUMS file
+func NewSha256SumsEntry(line string) (Sha256SumsEntry, error) {
+	r := regexp.MustCompile("\\s+")
+	s := r.Split(line, -1)
+	if len(s) != 2 {
+		return Sha256SumsEntry{}, fmt.Errorf("line contains %d parts instead of 2", len(s))
+	}
+
+	sum, err := hex.DecodeString(s[0])
+	if err != nil {
+		return Sha256SumsEntry{}, err
+	}
+
+	return Sha256SumsEntry{
+		Sum:      sum,
+		FileName: s[1],
+	}, nil
+}
+
+type Sha256Sums struct {
+	Entries  []Sha256SumsEntry
+	Filename string
+}
+
+// Name returns the name of the provider of the SHA256SUMS file
+func (s *Sha256Sums) Name() string {
+	// RegEx could fail in rare cases as the first capture group doesn't try to match as much as possible
+	r := regexp.MustCompile("^terraform-provider-(?P<name>.+)_(?P<version>.+)_SHA256SUMS$")
+	matches := r.FindStringSubmatch(s.Filename)
+	return matches[1]
```

Comment:
Do we want to check the length of matches as a precaution or check against nil? https://pkg.go.dev/regexp#Regexp.FindStringSubmatch

Reply:
Good point, I addressed this in  5c55979